src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -435,16 +410,34 @@ public override void GetConditionalDependenciesDueToEETypePresence(ref CombinedD
                 // of the bases/interfaces are annotated.
                 // ObjectGetTypeFlowDependencies don't need to be conditional in that case. They'll be added as needed.
             }
+
+            // Ensure fields can be consistently reflection set & get.
+            if (type.HasInstantiation && !type.IsTypeDefinition && !IsReflectionBlocked(type))
+            {
+                foreach (FieldDesc field in type.GetFields())
+                {
+                    // Tiny optimization: no get/set for literal fields
+                    if (field.IsLiteral)
+                        continue;
```

Comment:
So literal fields have special handling in reflection code? Meaning they don't need any runtime support, it's all done via metadata, right?
(and we assume that since it's literal its type is primitive/string and thus we don't need to explicitly preserve its type either).

Reply:
Literal fields (const/enum values) only exist in metadata. They don't have a "value" that is accessible through IL. Reflection GetValue reads the value from the metadata instead of reading some location in memory (like we do for other fields). There's no location in memory to worry about.